api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -888,37 +889,83 @@ public void updateFamilyName_badRequest() {
 
   @Test
   public void resendWelcomeEmail_messagingException() throws MessagingException {
+    config.access.tiersVisibleToUsers =
+        Arrays.asList(
+            AccessTierService.REGISTERED_TIER_SHORT_NAME,
+            AccessTierService.CONTROLLED_TIER_SHORT_NAME);
     createAccountAndDbUserWithAffiliation();
     dbUser.setFirstSignInTime(null);
     when(mockDirectoryService.resetUserPassword(anyString())).thenReturn(googleUser);
     doThrow(new MessagingException("exception"))
         .when(mockMailService)
-        .sendWelcomeEmail_deprecated(any(), any(), any());
+        .sendWelcomeEmail(any(), any(), any(), any(), anyBoolean(), anyBoolean());
 
     ResponseEntity<Void> response =
         profileController.resendWelcomeEmail(
             new ResendWelcomeEmailRequest().username(dbUser.getUsername()).creationNonce(NONCE));
     assertThat(response.getStatusCode()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
     // called twice, once during account creation, once on resend
-    verify(mockMailService, times(2)).sendWelcomeEmail_deprecated(any(), any(), any());
+    verify(mockMailService, times(2))
+        .sendWelcomeEmail(any(), any(), any(), any(), anyBoolean(), anyBoolean());
     verify(mockDirectoryService, times(1)).resetUserPassword(anyString());
   }
 
   @Test
   public void resendWelcomeEmail_OK() throws MessagingException {
+    config.access.tiersVisibleToUsers =
+        Arrays.asList(
+            AccessTierService.REGISTERED_TIER_SHORT_NAME,
+            AccessTierService.CONTROLLED_TIER_SHORT_NAME);
+
     createAccountAndDbUserWithAffiliation();
     when(mockDirectoryService.resetUserPassword(anyString())).thenReturn(googleUser);
-    doNothing().when(mockMailService).sendWelcomeEmail_deprecated(any(), any(), any());
+    doNothing()
+        .when(mockMailService)
+        .sendWelcomeEmail(any(), any(), any(), any(), anyBoolean(), anyBoolean());
 
     ResponseEntity<Void> response =
         profileController.resendWelcomeEmail(
             new ResendWelcomeEmailRequest().username(dbUser.getUsername()).creationNonce(NONCE));
     assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
     // called twice, once during account creation, once on resend
-    verify(mockMailService, times(2)).sendWelcomeEmail_deprecated(any(), any(), any());
+    verify(mockMailService, times(2))
+        .sendWelcomeEmail(any(), any(), any(), any(), anyBoolean(), anyBoolean());
     verify(mockDirectoryService, times(1)).resetUserPassword(anyString());
   }
 
+  @Test
+  public void sendUserInstructions_none_deprecated() throws MessagingException {
```

Comment:
these new tests are not correct.  We already have the "deprecated" versions.  We need the undeprecated versions.

Reply:
also need a "deleted instructions" undeprecated test

Reply:
Added a test to handle the un-deprecated case.